### ISSUE TITLE
asrestore: Unexpected character \x0a in backup block (line 4, col 46), expected " "

### ISSUE THREAD

**Issue by TvoroG** [aerospike/aerospike-tools-backup#36]
Hi!

I did backup of namespace with `asbackup 3.10.0` in `aerospike 6.0.0.1` and tried to restore with `asrestore 3.12.0` in `aerospike 6.1.0.3`.

Here is redacted output:
```
2022-10-10 10:22:16 GMT [INF] [26865] Starting restore to localhost:3000 (bins: [all], sets: [all]) from ...
2022-10-10 10:22:16 GMT [INF] [26865] Processing 3 node(s)
2022-10-10 10:22:16 GMT [INF] [26865] Restoring ...
2022-10-10 10:22:16 GMT [INF] [26865] Opened backup file ...
2022-10-10 10:22:17 GMT [INF] [26865] Restoring records
2022-10-10 10:22:17 GMT [ERR] [26942] Unexpected character \x0a in backup block (line 4, col 46), expected " "
2022-10-10 10:22:17 GMT [ERR] [26942] Error while restoring backup file ...
2022-10-10 10:22:17 GMT [INF] [26901] 0 UDF file(s), 0 secondary index(es), 0 record(s) (0 rec/s, 253 KiB/s, 0 B/rec, retries: 0)
2022-10-10 10:22:17 GMT [INF] [26901] Expired 0 : skipped 0 : err_ignored 0 : inserted 0: failed 0 (existed 0 , fresher 0)
2022-10-10 10:22:17 GMT [INF] [26901] 0% complete, ~1d4h6m38s remaining
```

I've successfully restored backup with `asrestore 3.10.0`.

**Comment by dwelch-spike** [comment]
Hi @TvoroG,

Sorry for the delay. If you still have the data, can you post a run of asrestore 3.12.0 trying to restore the backup file using the `--verbose` flag? This will give me some more information for looking into the issue. If possible, the backup and restore commands you are using, aswell as a snippet of the backup file data would be useful. I'll try to reproduce the problem on my end.

Thanks for reporting the issue.

**Comment by dwelch-spike** [comment]
I think this issue stemmed from a change in asbackup 3.12.0 that caused a space to be appended after each secondary index entry in the backup file. Asrestore 3.12.0 expects a space after the secondary indexes but because this space isn't present in older backup files, it fails. This should be fixed in asbackup 3.13.1. If this is still a problem, please reopen the ticket.
